frontend/src/api/api.js

```diff
@@ -0,0 +1,69 @@
+import debounce from "lodash.debounce";
+
+export async function getTodos() {
+  return await fetch('/todos').then(res => res.json())
+}
+
+export async function getTodo(id) {
+  return await fetch(`/todos/${id}`)
+    .then((response) => {
+      if (response.status === 200) {
+        return response.json();
+      }
+      throw new Error(`${response.status} - ${response.statusText}`);
+    })
+}
+
+export function createTodo(title, setTodos) {
+  fetch('/todos/create', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ title })
+  })
+    .then(res => res.json())
+    .then(res => setTodos(res))
+}
+
+export function updateTodo(todo, setTodo, setTodos) {
+  fetch(`/todos/update`, {
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ todo })
+  })
+    .then(res => res.json())
+    .then(res => setTodo(res))
+    .finally(() => this.getTodos().then(todos => setTodos(todos)));
```

Comment:
Have you tested this part of the code? When using `this` in an arrow function, it points to `window`, which will be translated to `window.getTodos`.